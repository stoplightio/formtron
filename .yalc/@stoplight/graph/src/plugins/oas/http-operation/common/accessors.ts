import unionBy = require('lodash/unionBy');

export function getOasParameters<ParamType extends { name: string; in: string }>(
  operationParameters: ParamType[] | undefined,
  pathParameters: ParamType[] | undefined
) {
  return unionBy(operationParameters, pathParameters, (parameter: ParamType) => `${parameter.name}-${parameter.in}`);
}
