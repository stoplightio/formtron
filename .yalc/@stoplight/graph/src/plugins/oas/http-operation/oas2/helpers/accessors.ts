import get = require('lodash/get');
import pick = require('lodash/pick');
import values = require('lodash/values');
import { Operation, Security, Spec } from 'swagger-schema-official';
import { uniqFlatMap } from '../../common/utils';

export function getSecurities(operation: Partial<Operation>, spec: Partial<Spec>): Security[] {
  const globalSecuritySchemes = spec.security || [];
  const schemes = uniqFlatMap(globalSecuritySchemes);
  const globalSecurity = values(pick(spec.securityDefinitions || {}, schemes));
  return operation.security || globalSecurity;
}

function getProducesOrConsumes(which: 'produces' | 'consumes', operation: Partial<Operation>, spec: Partial<Spec>) {
  const mimeTypes = get(operation, which, get(spec, which, [])) as string[];
  return mimeTypes.length ? mimeTypes : ['*'];
}

export function getProduces(operation: Partial<Operation>, spec: Partial<Spec>) {
  return getProducesOrConsumes('produces', operation, spec);
}

export function getConsumes(operation: Partial<Operation>, spec: Partial<Spec>) {
  return getProducesOrConsumes('consumes', operation, spec);
}
