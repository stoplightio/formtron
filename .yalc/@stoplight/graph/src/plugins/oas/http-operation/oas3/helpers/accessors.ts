import { get, pick, values } from 'lodash';
import { OpenAPIObject, OperationObject, SecuritySchemeObject } from 'openapi3-ts';
import { uniqFlatMap } from '../../common/utils';

export function getOas3Securities(
  operation: Partial<OperationObject>,
  spec: Partial<OpenAPIObject>
): SecuritySchemeObject[] {
  const globalSchemes = uniqFlatMap(spec.security);
  const operationSchemes = uniqFlatMap(operation.security);
  return values(pick(get(spec, 'components.securitySchemes'), operation.security ? operationSchemes : globalSchemes));
}
