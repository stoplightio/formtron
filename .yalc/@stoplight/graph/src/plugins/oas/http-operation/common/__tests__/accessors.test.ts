import { getOasParameters } from '../accessors';

describe('getOasParameters', () => {
  test('should return empty array', () => {
    expect(getOasParameters(undefined, undefined)).toEqual([]);
  });

  test('should fallback to operation parameters', () => {
    expect(getOasParameters([{ name: 'n1', in: 'i1' }, { name: 'n2', in: 'i2' }], undefined)).toMatchSnapshot();
  });

  test('should fallback to path parameters', () => {
    expect(getOasParameters(undefined, [{ name: 'n1', in: 'i1' }, { name: 'n2', in: 'i2' }])).toMatchSnapshot();
  });

  test('should prefer operation parameters', () => {
    expect(
      getOasParameters(
        [{ name: 'n1', in: 'n1', type: 'array' }, { name: 'no2', in: 'io2' }],
        [{ name: 'n1', in: 'n1', type: 'string' }, { name: 'np3', in: 'ip3' }]
      )
    ).toMatchSnapshot();
  });
});
