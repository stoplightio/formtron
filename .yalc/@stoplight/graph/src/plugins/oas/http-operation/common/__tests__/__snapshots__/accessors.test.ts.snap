// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOasParameters should fallback to operation parameters 1`] = `
Array [
  Object {
    "in": "i1",
    "name": "n1",
  },
  Object {
    "in": "i2",
    "name": "n2",
  },
]
`;

exports[`getOasParameters should fallback to path parameters 1`] = `
Array [
  Object {
    "in": "i1",
    "name": "n1",
  },
  Object {
    "in": "i2",
    "name": "n2",
  },
]
`;

exports[`getOasParameters should prefer operation parameters 1`] = `
Array [
  Object {
    "in": "n1",
    "name": "n1",
    "type": "array",
  },
  Object {
    "in": "io2",
    "name": "no2",
  },
  Object {
    "in": "ip3",
    "name": "np3",
  },
]
`;
